client/homebrew/pages/basePages/listPage/listPage.jsx

```diff
@@ -167,12 +169,13 @@ const ListPage = createClass({
 	},
 
 	renderSortOptions : function(){
+		''.setTranslationDefaults(['userpage', 'filters']);
 		return <div className='sort-container'>
-			<h6>Sort by :</h6>
-			{this.renderSortOption('Title', 'alpha')}
-			{this.renderSortOption('Created Date', 'created')}
-			{this.renderSortOption('Updated Date', 'updated')}
-			{this.renderSortOption('Views', 'views')}
+			<h6>{'sortyBy'.translate()}:</h6>
+			{this.renderSortOption('title'.translate(), 'alpha')}
+			{this.renderSortOption('created date'.translate(), 'created')}
+			{this.renderSortOption('updated date'.translate(), 'updated')}
+			{this.renderSortOption('views'.translate(), 'views')}
```

Comment:
Instead of translating each of these four options individually, you can instead place a `.translate()` on line ~125:

```js
	renderSortOption : function(sortTitle, sortValue){
		return <div className={`sort-option ${(this.state.sortType == sortValue ? 'active' : '')}`}>
			<button
				value={`${sortValue}`}
				onClick={this.state.sortType == sortValue ? this.handleSortDirChange : this.handleSortOptionChange}
			>
				{`${sortTitle.translate()}`}   //  <---------  add .translate() here
			</button>
			{this.state.sortType == sortValue &&
				<i className={`sortDir fas ${this.state.sortDir == 'asc' ? 'fa-sort-up' : 'fa-sort-down'}`}></i>
			}
		  </div>;
	},
```

Reply:
i am unfamiliar with this specific code, would we keep the same key names in the yaml files?

Reply:
Yeah, the yaml's will stay the same.  

Right now in the PR, the sort buttons are rendered in the `renderSortOption` function, and the actual text that displays on the buttons is done here:

```jsx
<button ...>
	{`${sortTitle}`}
</button>
```

`sortTitle` is an arg of `renderSortOption`, and is just a string.  In the PR, right now, the string that is being passed here into this arg is already translated.   But it doesn't have to be.  Instead, you can pass the original english string into `renderSortOption`.  Then, we can translate `sortTitle` (which is just the string we are passing in).

Just for completeness, here are the two entire functions as I describe them:
```jsx
	renderSortOption : function(sortTitle, sortValue){
		return <div className={`sort-option ${(this.state.sortType == sortValue ? 'active' : '')}`}>
			<button
				value={`${sortValue}`}
				onClick={this.state.sortType == sortValue ? this.handleSortDirChange : this.handleSortOptionChange}
			>
				{`${sortTitle.translate()}`}
			</button>
			{this.state.sortType == sortValue &&
				<i className={`sortDir fas ${this.state.sortDir == 'asc' ? 'fa-sort-up' : 'fa-sort-down'}`}></i>
			}
		  </div>;
	},

// ...

	renderSortOptions : function(){
		''.setTranslationDefaults(['userPage', 'filters']);
		return <div className='sort-container'>
			<h6>{'sortBy'.translate()}:</h6>
			{this.renderSortOption('title', 'alpha')}
			{this.renderSortOption('created date', 'created')}
			{this.renderSortOption('updated date', 'updated')}
			{this.renderSortOption('views', 'views')}
			{/* {this.renderSortOption('Latest', 'latest')} */}

			{this.renderFilterOption()}
		</div>;
	},
```

(*note, both of these have very similar function names....possibly that should be addressed but I'm not concerned about it now*)